test/e2e/upgradetest/upgrade_test.go

```diff
@@ -0,0 +1,58 @@
+// Copyright 2017 The etcd-operator Authors
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package upgradetest
+
+import (
+	"testing"
+)
+
+func TestOperatorUpgrade(t *testing.T) {
```

Comment:
can we create different test for different cases? i am less concerned about the duplication, more concerned about the debugability. separating test allows us to isolate and run only one case.